### PR TITLE
Exporters: Reload service if user has changed attributes

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
verified that it works:
```
Info: Applying configuration version '1594364693'
Notice: /Stage[main]/Prometheus::Bird_exporter/Prometheus::Daemon[bird_exporter]/User[bird-exporter]/groups: groups changed bird-exporter to ['bird', 'bird-exporter', 'bird6'] (corrective)
Info: /Stage[main]/Prometheus::Bird_exporter/Prometheus::Daemon[bird_exporter]/User[bird-exporter]: Scheduling refresh of Service[bird_exporter]
Notice: /Stage[main]/Prometheus::Bird_exporter/Prometheus::Daemon[bird_exporter]/Service[bird_exporter]: Triggered 'refresh' from 1 event
Notice: Applied catalog in 43.25 seconds
```

**Review by mmerfort — APPROVED** [review]

File: manifests/daemon.pp [hunk 1]
```diff
@@ -135,11 +135,19 @@
     }
   }
   if $manage_user {
+    # if we manage the service, we need to reload it if our user changes
+    # important for cases where another group gets added
+    if $manage_service {
+      $notify = $notify_service
+    } else {
+      $notify = undef
+    }
```
Comment (ekohl) [inline manifests/daemon.pp:144]: Untested, but wouldn't it be better to use:
```puppet
if $manage_service {
  User[$user] ~> $notify_service
}
```

Comment (bastelfreak) [reply manifests/daemon.pp:144]: mhm, that could work as well

Comment (bastelfreak) [reply manifests/daemon.pp:144]:
```
# lid  bird-exporter
 bird(gid=484)
 bird6(gid=483)
 bird-exporter(gid=482)
# usermod -G bird-exporter bird-exporter
# lid  bird-exporter
 bird-exporter(gid=482)
# puppet agent -t --environment testenv
Info: Using configured environment 'testenv'
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Retrieving locales
Info: Loading facts
Info: Caching catalog for testnode
Info: Applying configuration version '1595330160'
Notice: /Stage[main]/Prometheus::Bird_exporter/Prometheus::Daemon[bird_exporter]/User[bird-exporter]/groups: groups changed bird-exporter to ['bird', 'bird-exporter', 'bird6'] (corrective)
Info: /Stage[main]/Prometheus::Bird_exporter/Prometheus::Daemon[bird_exporter]/User[bird-exporter]: Scheduling refresh of Service[bird_exporter]
Notice: /Stage[main]/Prometheus::Bird_exporter/Prometheus::Daemon[bird_exporter]/Service[bird_exporter]: Triggered 'refresh' from 1 event
Notice: Applied catalog in 42.21 seconds
#
```

I will create a PR

**Pull request merged**